frontend/packages/gitops-plugin/src/components/details/GitOpsDetails.tsx

```diff
@@ -38,63 +97,112 @@ const GitOpsDetails: React.FC<GitOpsDetailsProps> = ({ envs, appName }) => {
+                          </Tooltip>
+                        </h2>
                       </StackItem>
-                    </Stack>
-                  </CardTitle>
-                  <CardBody>
-                    <Stack>
                       <StackItem className="co-truncate co-nowrap">
                         {env.cluster ? (
                           <ExternalLink
-                            additionalClassName="odc-gitops-details__env-section__url"
+                            additionalClassName="odc-gitops-details__env-section__cluster-url"
                             href={env.cluster}
                           >
-                            <Label
-                              className="odc-gitops-details__env-section__url-label"
-                              isTruncated
-                            >
-                              {env.cluster}
-                            </Label>
+                            {env.cluster}
                           </ExternalLink>
                         ) : (
-                          <div className="odc-gitops-details__env-section__url-empty-state">
+                          <div className="odc-gitops-details__env-section__cluster-url-empty-state">
                             {t('gitops-plugin~Cluster URL not available')}
                           </div>
                         )}
                       </StackItem>
-                      <StackItem className="co-truncate co-nowrap">
-                        <span className="co-resource-item odc-gitops-details__env-section__co-resource-item">
-                          <ResourceIcon kind="Project" />
-                          <Label className="co-resource-item__resource-name" isTruncated>
-                            {env.environment}
-                          </Label>
-                        </span>
-                      </StackItem>
-                      {env.environment && argocdLink && (
-                        <StackItem className="co-truncate co-nowrap">
-                          <ExternalLink
-                            href={`${argocdLink.spec.href}/applications/${env.environment}-${appName}`}
-                            text="Argo CD"
-                            additionalClassName="odc-gitops-details__env-section__argocd-link"
-                          />
+                      {env.status && (
+                        <StackItem className="odc-gitops-details__env-section__status-label">
+                          <Tooltip content="Sync status">
+                            <span>{renderStatusLabel(env.status)}</span>
+                          </Tooltip>
+                        </StackItem>
+                      )}
+                    </Stack>
+                  </CardTitle>
+                  <CardBody>
+                    <Stack className="odc-gitops-details__revision">
+                      {env.revision ? (
+                        <>
+                          {env.revision?.message && (
```

Comment:
`revision` must be on `env` based on your condition 2 lines up

```suggestion
                          {env.revision.message && (
```